components/script/dom/htmlinputelement.rs

```diff
@@ -1016,11 +1052,25 @@ impl HTMLInputElement {
         }
 
         // Step 5.12
-        vec![FormDatum {
+        let mut result = vec![FormDatum {
             ty: ty.clone(),
             name: name,
             value: FormDatumValue::String(self.Value()),
-        }]
+        }];
+
+        // 4.10.18.2
+        // https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#submitting-element-directionality:-the-dirname-attribute
+        let dirname: DOMString = self.DirName();
+        let directionality = DOMString::from(self.directionality("auto"));
```

Comment:
Right now this only checks the dirname for input elements. We should move the dirname processing into https://github.com/servo/servo/blob/f5c086b30759e7203bb1ab412693f0e86252dafc/components/script/dom/htmlformelement.rs#L651 and implement step 13 of https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#constructing-the-form-data-set so it affects all form elements.